src/components/org/apache/jmeter/extractor/json/jsonpath/JSONPostProcessor.java

```diff
@@ -0,0 +1,236 @@
+    private static final Logger log = LoggingManager.getLoggerForClass();
+    private static final String JSON_PATH_EXPRESSIONS = "ULP.json.jsonPathExprs"; // $NON-NLS-1$
+    private static final String REFERENCE_NAMES = "ULP.json.referenceNames"; // $NON-NLS-1$
+    private static final String DEFAULT_VALUES = "ULP.json.defaultValues"; // $NON-NLS-1$
+    private static final String MATCH_NUMBERS = "ULP.json.match_numbers"; // $NON-NLS-1$
+    private static final String COMPUTE_CONCATENATION = "ULP.json.compute_concat"; // $NON-NLS-1$
+    private static final String REF_MATCH_NR = "_matchNr"; // $NON-NLS-1$
+    private static final String ALL_SUFFIX = "_ALL"; // $NON-NLS-1$
+    private static final String SEPARATOR = ";"; // $NON-NLS-1$
+    public static final boolean COMPUTE_CONCATENATION_DEFAULT_VALUE = false;
+    
+    private static final ThreadLocal<JSONManager> localMatcher = new ThreadLocal<JSONManager>() {
+        @Override
+        protected JSONManager initialValue() {
+            return new JSONManager();
+        }
+    };
+    public void process() {
+        JMeterContext context = getThreadContext();
+        JMeterVariables vars = context.getVariables();
+        String jsonResponse = "";
+        if (isScopeVariable()){ 
+            jsonResponse = vars.get(getVariableName());
+            if(log.isDebugEnabled()) {
+                log.debug("JSON Extractor is using variable:"+getVariableName()+" which content is:"+jsonResponse);
+            }
+        } else {
+            SampleResult previousResult = context.getPreviousResult();
+            if (previousResult == null) {
+                return;
+            }
+            jsonResponse = previousResult.getResponseDataAsString();
+            if(log.isDebugEnabled()) {
+                log.debug("JSON Extractor processing result:"+jsonResponse);
+            }
+        }
+        String[] refNames = getRefNames().split(SEPARATOR);
+        String[] jsonPathExpressions = getJsonPathExpressions().split(SEPARATOR); 
+        String[] defaultValues = getDefaultValues().split(SEPARATOR);
+        int[] matchNumbers = getMatchNumbersAsInt(defaultValues.length);
+
+        //jsonResponse = jsonResponse.replaceAll("'", "\""); // $NON-NLS-1$  $NON-NLS-2$
+
+        if (refNames.length != jsonPathExpressions.length || 
+                refNames.length != defaultValues.length) {
+            log.error("Number of JSON Path variables must match number of default values and json-path expressions, check you use separator ';' if you have many values"); // $NON-NLS-1$
+            throw new IllegalArgumentException(JMeterUtils.getResString("jsonpp_error_number_arguments_mismatch_error")); // $NON-NLS-1$
+        } else {
+            for (int i = 0; i < jsonPathExpressions.length; i++) {
+                int matchNumber = matchNumbers[i];
+                String refNameFromRefNames = refNames[i].trim();
+                String jsonPathFromPathes = jsonPathExpressions[i].trim();
```

Comment:
Plural of path is paths.  I would go with something like path, jsonPath or current(Json)?Path